### DISCUSSION TITLE
Boost Event

### DISCUSSION THREAD

**Discussion by GitMilchi** [discord-jda/JDA#2559] [Questions and Help]
### General Troubleshooting

- [X] I have checked for similar issues on the Issue-tracker.
- [X] I have updated to the [latest JDA version](https://ci.dv8tion.net/job/JDA/lastSuccessfulBuild/)
- [X] I have checked the branches or the maintainers' PRs for upcoming features.

### Feature Request

Please give me an example how to send messages when a member is boosting a guild. But I only want if the member adds a boost and not when the boost expires.

**Comment by kazuryyx** [comment]
There is [GuildMemberUpdateBoostTimeEvent](https://docs.jda.wiki/net/dv8tion/jda/api/events/guild/member/update/GuildMemberUpdateBoostTimeEvent.html).

The event triggers on both boost and boost expiry. You may be able to check if the new time of the event is null, therefore the boost ended (_?, correct me if wrong_)
And then when the check is true, you can just simply get a text channel by id and send a message to that channel that the member has now boosted.

**Comment by Andre601** [comment]
From what I know can you use the `getNewTimeBoosted()` from the previously mentioned event.

It's nullable so by logic would it be null if the member sttops boosting.

Alternatively listen for a [`GuildMemberRoleAddEvent`](https://docs.jda.wiki/net/dv8tion/jda/api/events/guild/member/GuildMemberRoleAddEvent.html) and check if (one of) the added role(s) is the booster role ([`Role#getTags().isBoost()`](https://docs.jda.wiki/net/dv8tion/jda/api/entities/Role.RoleTags.html#isBoost())).
